Tasks/Maven/maventask.ts

```diff
@@ -301,22 +320,19 @@ function uploadBuildArtifactsFromResult(analysisResult:ar.AnalysisResult):void {
     // Copy files to a staging folder and upload the entire folder
     // This is a workaround until the following bug is fixed: https://github.com/Microsoft/vso-agent/issues/263
     filesToUpload.forEach(fileToUpload => {
-        var stagingFilePath = path.join(stagingDir, path.basename(fileToUpload));
+        var stagingFilePath = path.join(toolStagingDir, path.basename(fileToUpload));
         tl.debug('Staging ' + fileToUpload + ' to ' + stagingFilePath);
         // Execute the copy operation. -f overwrites if there is already a file at the destination.
         tl.cp('-f', fileToUpload, stagingFilePath);
     });
 
-    console.info('Uploading artifacts for ' + analysisResult.toolName);
+    console.log('Uploading artifacts for ' + analysisResult.toolName + ' from ' + toolStagingDir);
 
     tl.command("artifact.upload", {
         containerfolder: 'codeAnalysis',
-        artifactname: analysisResult.toolName
-    }, stagingDir);
-
-    // clean up afterwards
-    tl.rmRF(stagingDir);
-    tl.mkdirP(stagingDir);
```

Comment:
These operations - tl.rmRF() and tl.mkdirP() - were the cause of the async issues.